src/components/AssetHeader/AssetHeader.tsx

```diff
@@ -70,14 +70,13 @@ export const AssetHeader: React.FC<AssetHeaderProps> = ({ assetId, accountId })
 
   const href = (() => {
     const { assetReference } = fromAssetId(asset.assetId)
-    const maybeToken = tokenOrUndefined(assetReference)
 
     if (isNft(asset.assetId)) {
       const [token] = assetReference.split('/')
       return `${asset.explorer}/token/${token}?a=${asset.id}`
     }
 
-    if (maybeToken) return `${asset?.explorerAddressLink}${maybeToken}`
+    if (isToken(asset.assetId)) return `${asset?.explorerAddressLink}${assetReference}`
```

Comment:
Noice! Actually prefer this one over using `contractAddressOrUndefined` in l.73, more declarative.